src/Database/TigerBeetle/Raw/Client.hs

```diff
@@ -74,6 +85,98 @@ data ClientState = ClientState
 
 newtype ClientHandle = ClientHandle { tvar :: TVar ClientState }
 
+data Address = Address { getAddress :: Text }
+  deriving (Eq, Show)
+
+toCString :: Address -> ((Ptr CChar, Int) -> IO a) -> IO a
+toCString = BS.useAsCStringLen . TE.encodeUtf8 . getAddress
```

Comment:
maybe we should call this something like `withAddressPtr`, its a convention I try to use when naming these "bracket-like" functions that control resources via a callback